packages/core/src/parseSamenApp.ts

```diff
@@ -27,15 +28,31 @@ export interface ParsedSamenFunctionDefinition {
   parameters: ts.ParameterDeclaration[]
   returnType: ts.TypeNode
   config: ParsedSamenFunctionConfig
+  context?: {
+    name: string
+    type: ts.TypeNode
+  }
 }
 
-export interface ParsedSamenFunctionConfig {
+export interface ParsedSamenServiceConfig {
   memory?: number
   timeout?: number
 
   minInstance?: number
   maxInstance?: number
-  middleware?: ts.FunctionLikeDeclarationBase[]
+  middleware?: ParsedMiddlewareConfig[]
```

Comment:
I think we should be using "context" as a consistent concept through the whole stack, instead of "middleware" in parts of it. This could be a `contextProviders` property for example. What do you think?

Reply:
While building this I realised that `middleware` and `context` are 2 separate constructs. It's perfectly possible to write middleware which doesn't "provide" any context. And also to provide context without the need for any middleware. IMO middleware is the correct term here.

Reply:
Ah, fair enough! 👍